test/test_python_dispatch.py

```diff
@@ -523,5 +523,51 @@ def wrap(e):
         self.assertEqual(x.grad, y)
         self.assertEqual(y.grad, x)
 
+    def test_none_wrapping(self):
+        # A Tensor subclass that returns None when doing add
+        # See LoggingTensor above for more details on the subclass
+        class SublassWithNone(torch.Tensor):
+            @staticmethod
+            def __new__(cls, elem, *args, **kwargs):
+                r = torch.Tensor._make_wrapper_subclass(
+                    cls, elem.size(),
+                    dtype=elem.dtype, layout=elem.layout,
+                    device=elem.device, requires_grad=elem.requires_grad
+                )
+                r.elem = elem
+                return r
+
+            @classmethod
+            def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
+                def unwrap(e):
+                    return e.elem if isinstance(e, SublassWithNone) else e
+
+                def wrap(e):
+                    return SublassWithNone(e) if isinstance(e, torch.Tensor) else e
+
+                # no_dispatch is only needed if you use enable_python_mode.
+                # It prevents infinite recursion.
+                with no_dispatch():
+                    rs = tree_map(wrap, func(*tree_map(unwrap, args), **tree_map(unwrap, kwargs)))
+                if func.__name__ == "add":
+                    return None
```

Comment:
I guess we're saved by no native functions returning `optional<Tensor>`...

Reply:
Not sure what you mean?
The optional<Tensor> case is handled differently in the caster code: https://github.com/pytorch/pytorch/blob/da580fe02ed74074d94252949adaf1aaa4bb939d/torch/csrc/jit/python/pybind_utils.cpp#L186-L193

Reply:
If some native functions' schema show them as returning `optional<Tensor>` and some show them as returning `Tensor`, I would be scared if the ones that return `Tensor` can also return None